components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1010,33 +1010,33 @@ public class MonitoredItemWrapper {
             /// <summary>
             /// Cache of the latest events for the pending alarms optionally monitored
             /// </summary>
-            public ConcurrentDictionary<string, MonitoredItemNotificationModel> PendingAlarmEvents { get; } = new ConcurrentDictionary<string, MonitoredItemNotificationModel>();
+            public Dictionary<string, MonitoredItemNotificationModel> PendingAlarmEvents { get; } = new Dictionary<string, MonitoredItemNotificationModel>();
```

Comment:
What is the rational behind not to use the ConcurrentDictionary?

Reply:
We have two primitives (the collection and the dirty flag) that needs to be changed atomically. Since I needed to do that and saw no other alternative than to use a lock - using ConcurrentDictionary while under exclusive lock seemed unneccessary

Reply:
@koepalex?

Reply:
You are right, if the exclusive lock is required you don't need the ConcurrentDictionary, my intended questions was more 
why not only use ConcurrentDictionary and avoid the lock